apps/aecore/src/aec_db_gc.erl

```diff
@@ -0,0 +1,206 @@
+-module(aec_db_gc).
+
+-behaviour(gen_statem).
+
+%% API
+-export([start_link/0,
+         start_link/3,
+         maybe_garbage_collect/0,
+         stop/0]).
+
+%% gen_statem callbacks
+-export([callback_mode/0, init/1, terminate/3, code_change/4]).
+-export([handle_event/4]).
+
+-record(data,
+        {enabled  :: boolean(),                         % do we garbage collect?
+         interval :: non_neg_integer(),                 % how often (every `interval` blocks) GC runs
+         history  :: non_neg_integer(),                 % how many block state back from top to keep
+         synced   :: boolean(),                         % we only run GC if chain is synced
+         height   :: undefined | non_neg_integer(),     % latest height of MPT hashes stored in tab
+         hashes   :: undefined | pid() | reference()}). % hashes tab (or process filling the tab 1st time)
```

Comment:
The name `hashes` is misleading. I'd expect a list here, but its actually a reference (to a process or table).

Reply:
Matter of preference really :).. I understand the plural indicates any collection (not just list) OR means to create such collection.

Reply:
But in the possible cases the API to interact with the value is different, one is an ETS table, the other one a process. I'm not going to block because of this, but this is clearly an overloaded semantic.